script.js

```diff
@@ -20,24 +20,18 @@ setByteLengthPerOrigin = (origin, byteLength) => {
 }
 
 headersReceivedListener = (requestDetails) => {
-  let filter = browser.webRequest.filterResponseData(requestDetails.requestId);
-
-  filter.ondata = event => {
-    const origin = extractHostname(!requestDetails.originUrl ? requestDetails.url : requestDetails.originUrl);
-    setByteLengthPerOrigin(origin, event.data.byteLength);
-
-    filter.write(event.data);
-  };
-
-  filter.onstop = () => {
-    filter.disconnect();
-  };
+     const origin = extractHostname(!requestDetails.initiator ? (!requestDetails.originUrl ? requestDetails.url : requestDetails.originUrl) : requestDetails.initiator);
+     const rhf = requestDetails.responseHeaders.find(element => element.name.toLowerCase() === "content-length");
+     const rh = undefined === rhf ? {value: 0} 
+      : rhf;
```

Comment:
Please use significant variable. requestDetails instead of rh, contentLengthHeader instead rhf ?

Reply:
Done, thanks for the feedback ;)